provisioner/unpacker/main.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"archive/tar"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+type unpackOptions struct {
+	BundleName       string
+	StorageNamespace string
+	ManifestDir      string
+}
+
+func newCmd() *cobra.Command {
+	o := unpackOptions{}
+
+	cmd := &cobra.Command{
+		Use: "run",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			logger := logrus.New()
+			if err := o.run(ctx, logger); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	cmd.Flags().StringVar(&o.BundleName, "bundle", "", "configures which existing Bundle custom resource to pull and unpack referenced content")
+	cmd.Flags().StringVar(&o.StorageNamespace, "namespace", "k8s-provisioner", "configures the namespace that houses the bundle storage resources")
+	cmd.Flags().StringVar(&o.ManifestDir, "manifests", "manifests", "configures the directory that contains the plain bundle manifests to unpack")
+
+	return cmd
+}
+
+func main() {
+	cmd := newCmd()
+	if err := cmd.Execute(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func (o unpackOptions) run(ctx context.Context, logger logrus.FieldLogger) error {
+	config, err := controllerruntime.GetConfig()
+	if err != nil {
+		return err
+	}
+	scheme := runtime.NewScheme()
+	if err := v1alpha1.AddToScheme(scheme); err != nil {
+		return err
+	}
+	if err := corev1.AddToScheme(scheme); err != nil {
+		return err
+	}
+	client, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+
+	bundle := &v1alpha1.Bundle{}
+	if err := client.Get(ctx, types.NamespacedName{Name: o.BundleName}, bundle); err != nil {
+		return err
+	}
+
+	cm := &corev1.ConfigMap{}
+	if err := client.Get(ctx, types.NamespacedName{
```

Comment:
Here we get a configmap, but we don't check to see if it has content in it before attempting to update it later on. Does the unpacker ever care about overwriting some data already in storage?

Reply:
> Does the unpacker ever care about overwriting some data already in storage?

I've typically landed on the Bundle's source reference acts as the source-of-truth vs. what's present on-cluster, so I'm not super worried about overwriting the on-cluster representation if the end goal is a declarative installation. There's likely a caching layer that could be introduced to alleviate re-processing already present resources on-cluster, but I think that's something we can investigate further down the line as we work towards getting something worker vs. pre-maturely optimizing.